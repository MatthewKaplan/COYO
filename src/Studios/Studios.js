import React, { Component } from "react";
import { StudioCard } from "../StudioCard/StudioCard.js";
import "../Styles/_studioCard.scss";

export default class Studios extends Component {
  studiosToRender = path => {
    return path.map(studio => (
      <StudioCard
        key={studio.id}
        image={studio.img}
        name={studio.name}
        address={studio.address}
        phone={studio.phoneNum}
        mission={studio.mission}
        rating={studio.googleRating}
        roundedRating={`${Math.round(((studio.googleRating / 5) * 100) / 10) *
          10}%`}
        link={studio.link}
        fee={studio.dropInFee}
      />
    ));
  };

  render() {
    return (
      <section className="studioCardContainer">
        {this.studiosToRender(
          this.props.rendered.length >= 1
            ? this.props.rendered
            : this.props.studios
        )}
      </section>
    );
  }
}
