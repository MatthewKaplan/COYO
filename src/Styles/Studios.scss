@import 'Mixins';

.studioCardContainer {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-gap: 50px;
  margin-top: 100px;
  padding: 20px;
  width: 100%;
}