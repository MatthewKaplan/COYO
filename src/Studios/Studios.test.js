import React from 'react';
import Studios from './Studios';
import { shallow } from 'enzyme';
import MockData from "../Helper/mockData";

const mockRendered = MockData.mockStudios;
const mockStudios = MockData.mockStudios;
const mockRender = jest.fn()

describe('Studios', () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(
      <Studios 
        renderDisplay={mockRender}
        rendered={mockRendered}
        studios={mockStudios}
      />
    )
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});