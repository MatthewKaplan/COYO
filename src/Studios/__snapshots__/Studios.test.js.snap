// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Studios should match snapshot 1`] = `
<section
  className="studioCardContainer"
>
  <StudioCard
    address="1735 E 17th Ave, Denver, CO 80218"
    fee="$17.00"
    image="https://i.imgur.com/w7KfQWv.jpg"
    link="https://wholeyoga.com"
    mission="Where body and mind become whole."
    name="Whole Yoga"
    phone="(303)333-9642"
    rating={4.8}
    roundedRating="100%"
  />
  <StudioCard
    address="3101 Walnut St, Denver, CO 80205"
    fee="Suggested donation"
    image="https://i.imgur.com/nKIbWP2.jpg"
    link="https://www.rinoyogasocial.com/"
    mission="At RYS, our mission is simple: make yoga accessible for all."
    name="Rino Yoga Social"
    phone="(720)724-8393"
    rating={5}
    roundedRating="100%"
  />
  <StudioCard
    address="639 E 19th Ave, Denver, CO 80203"
    fee="$20.00"
    image="https://i.imgur.com/fkUvUxr.jpg"
    link="https://samadhiyoga.net"
    mission="To honor the science of yoga as a methodical process to create a lifestyle of health, happiness, and abundance."
    name="Samadhi Yoga"
    phone="(303)860-9642"
    rating={4.7}
    roundedRating="90%"
  />
</section>
`;
